File: new_server/vueblog-after/src/main/java/cn/smilex/vueblog/after/controller/UserController.java
package cn.smilex.vueblog.after.controller;

import cn.smilex.vueblog.common.annotation.CrossOrigin;
import cn.smilex.vueblog.common.entity.common.Result;
import cn.smilex.vueblog.common.service.UserService;
import cn.smilex.vueblog.common.util.CommonUtil;
import com.linecorp.armeria.common.HttpRequest;
import com.linecorp.armeria.server.annotation.*;
import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

import java.util.Optional;

/**
 * @author smilex
 * @date 2022/12/4/15:49
 * @since 1.0
 */
@SuppressWarnings("unused")
@Slf4j
@PathPrefix("/user")
@RequestConverter(JacksonRequestConverterFunction.class)
@ProducesJson
@CrossOrigin
@Component
public class UserController {
    private UserService userService;

    @Autowired
    public void setUserService(UserService userService) {
        this.userService = userService;
    }

    /**
     * 获取当前token中的用户信息
     *
     * @param request 请求对象
     * @return 用户信息
     */
    @Options("/info")
    @Post("/info")
    public Result<?> info(HttpRequest request) {
        return CommonUtil.tryRun(
                () -> userService.getUserInfoByToken(request),
                Optional.of(
                        e -> Optional.of(e.getMessage())
                )
        );
    }
}

File: new_server/vueblog-after/src/main/java/cn/smilex/vueblog/after/controller/UserController.kt
package cn.smilex.vueblog.after.controller

import cn.smilex.vueblog.common.annotation.CrossOrigin
import cn.smilex.vueblog.common.entity.common.Result
import cn.smilex.vueblog.common.handler.AuthService
import cn.smilex.vueblog.common.service.UserService
import com.linecorp.armeria.common.HttpRequest
import com.linecorp.armeria.server.annotation.*
import org.springframework.stereotype.Component

/**
 * @author smilex
 * @date 2023/12/3 17:35:46
 */
@Suppress("unused")
@PathPrefix("/user")
@RequestConverter(JacksonRequestConverterFunction::class)
@Decorator(AuthService::class)
@ProducesJson
@CrossOrigin
@Component
class UserController(private val userService: UserService) {

    /**
     * 获取当前token中的用户信息
     *
     * @param request 请求对象
     * @return 用户信息
     */
    @Post("/info")
    @Options("/info")
    fun info(request: HttpRequest): Result<*> {
        return Result.success(userService.getUserInfoByToken(request))
    }

}